samples/drm/app/main.js

```diff
@@ -0,0 +1,374 @@
+'use strict';
+
+angular.module('DashSourcesService', ['ngResource']).
+    factory('Sources', function($resource){
+        return $resource('app/sources.json', {}, {
+            query: {method:'GET', isArray:false}
+        });
+    });
+
+angular.module('DashNotesService', ['ngResource']).
+    factory('Notes', function($resource){
+        return $resource('app/notes.json', {}, {
+            query: {method:'GET', isArray:false}
+        });
+    });
+
+angular.module('DashContributorsService', ['ngResource']).
+    factory('Contributors', function($resource){
+        return $resource('app/contributors.json', {}, {
+            query: {method:'GET', isArray:false}
+        });
+    });
+
+angular.module('DashPlayerLibrariesService', ['ngResource']).
+    factory('PlayerLibraries', function($resource){
+        return $resource('app/player_libraries.json', {}, {
+            query: {method:'GET', isArray:false}
+        });
+    });
+
+angular.module('DashShowcaseLibrariesService', ['ngResource']).
+    factory('ShowcaseLibraries', function($resource){
+        return $resource('app/showcase_libraries.json', {}, {
+            query: {method:'GET', isArray:false}
+        });
+    });
+
+var app = angular.module('DashPlayer', [
+    'DashSourcesService',
+    'DashNotesService',
+    'DashContributorsService',
+    'DashPlayerLibrariesService',
+    'DashShowcaseLibrariesService'
+]);
+
+app.controller('DashController', function($scope, Sources, Notes, Contributors, PlayerLibraries, ShowcaseLibraries) {
+    var player,
+        video,
+        context;
+
+    $scope.drmData = [];
+
+    $scope.safeApply = function(fn) {
+        var phase = this.$root.$$phase;
+        if(phase == '$apply' || phase == '$digest')
+            this.$eval(fn);
+        else
+            this.$apply(fn);
+    };
+
+    ////////////////////////////////////////
+    //
+    // Error Handling
+    //
+    ////////////////////////////////////////
+
+    function onError(e) {
+
+    }
+
+    ////////////////////////////////////////
+    //
+    // Player Setup
+    //
+    ////////////////////////////////////////
+
+    video = document.querySelector(".dash-video-player video");
+    context = new Dash.di.DashContext();
+    player = new MediaPlayer(context);
+    $scope.version = player.getVersion();
+
+    player.startup();
+    player.addEventListener(MediaPlayer.events.ERROR, onError.bind(this));
+
+    player.attachView(video);
+    player.setAutoPlay(true);
+
+    var system = player.getSystem(),
```

Comment:
@KozhinM @AkamaiDASH Applications that use our player will need a high level of visibility into the EME process.  They will need many (if not all) of the events that are sent by ProtectionModel to verify that they are complying with the requirements mandated by premium content providers.

That being said, I completely agree with both of you.  The changes I made expose way too much of the player internals.  However, I think we need to make EME operations/events available outside the player.  I also think it is very valuable to provide access to a well-defined manifest object that can be downloaded and parsed by the player.  At this time, our manifest object is not very well defined and relies on the DashAdapter (which, as Mikhail said, might be going away soon).  I will add that as a feature request for a future release.

I am going to close this PR and suggest a new set of changes.  As part of this new PR, I will propose a new mechanism of exposing EME and the manifest to the application that does not involve so many player internals.  Additionally, I will try a new method of incorporating the DRM demo app into the index.html player so that I do not have to duplicate all of the assets and libraries.

Thank you for your feedback.